docs/chymyst01.md

```diff
@@ -386,43 +395,164 @@ Sometimes it appears that repeating input molecules is the most natural way of e
 However, I believe it is always possible to introduce some new auxiliary molecules and to rewrite the “chemistry laws” so that input molecules are not repeated while the resulting computations give the same results.
 This limitation could be lifted in a later version of `JoinRun` if it proves useful to do so.
 
-## Order of reactions
+## Order of reactions and nondeterminism
 
-When there are several different reactions that can start the available molecules, the runtime engine will choose the reaction at random.
-In the current implementation of `JoinRun`, the runtime will choose reactions at random, so that every reaction has an equal chance of starting.
+When there are several different reactions that can start the available molecules, the runtime engine will choose the reaction at random,
+so that every reaction has an equal chance of starting.
 
 Similarly, when there are several copies of the same molecule that can be consumed as input by a reaction, the runtime engine will make a choice of which copy 
 of the molecule to consume.
-Currently, `JoinRun` will _not_ randomize the input molecules but make an implementation-dependent choice.
+Currently, `JoinRun` will _not_ fully randomize the input molecules but make an implementation-dependent choice.
 A truly random selection of input molecules may be implemented in the future.
 
-It is not possible to assign priorities to reactions or to molecules.
+Importantly, it is _not possible_ to assign priorities to reactions or to molecules.
 The order of reactions in a join definition is ignored, and the order of molecules in the input list is also ignored.
-The debugging facility will print the molecule names in alphabetical order, and reactions will be printed in an unspecified order.
+Just for the purposes of debugging, molecules will be printed in alphabetical order of names, and reactions will be printed in an unspecified order.
```

Comment:
You must know that alphabet is not perfectly internationalized, but I don't worry about it. English language is huge market by itself.